### PR TITLE
[launch] fix accessing mount setting (Fixes #2157)

### LINKED ISSUE #2157: Auto-mount in primary fails when local.privileged-mounts is unreadable
I got this with a local build (with the settings file in `/root/.config/multipassd/...`):

```
$ multipass shell
Launched: primary
[2021-07-02T12:11:18.610] [error] [client] Caught an unhandled exception: Unable to read settings: access error (consider running with an administrative role)
```

Observed on Linux. Did not try other OSes.

**To Reproduce**
1. `multipass start`

**Expected behavior**
The launch would go through, with the mount being effected or skipped as appropriate.

### PR DESCRIPTION
**Pull request by Saviq**

### CODE REVIEW

**Review by townsend2010 — COMMENTED** [review]
This looks fine to me.  CI integration successfully completed on Hyperkit too: https://github.com/canonical/multipass/runs/3002450791?check_suite_focus=true

Other than seeing if coverage can be improved, I'm good with this.

File: src/client/cli/cmd/launch.cpp [hunk 1]
```diff
@@ -125,32 +125,16 @@ mp::ReturnCode cmd::Launch::run(mp::ArgParser* parser)
...
     auto ret = request_launch(parser);
     if (ret == ReturnCode::Ok && request.instance_name() == petenv_name.toStdString())
     {
-        if (!MP_SETTINGS.get_as<bool>(mounts_key))
-        {
-            cout << fmt::format("Skipping '{}' mount due to disabled mounts feature\n", mp::home_automount_dir);
-        }
-        else
+        std::string mounts_value;
+        if (std::tie(ret, mounts_value) = request_mounts_setting_from_daemon(parser); ret == ReturnCode::Ok)
```
Comment (townsend2010) [inline src/client/cli/cmd/launch.cpp:132]: This is really cool!  I never knew that could be done :smile:

**Review by townsend2010 — APPROVED** [review]
LGTM

bors merge

File: src/client/cli/cmd/launch.cpp (hunk 1 above)
Comment (ricab) [reply src/client/cli/cmd/launch.cpp:132]: Yeah, it was used mostly before structured bindings, the downside is that the variables need to be already declared :smiley:

**Pull request merged**